Unable to compile since recent `features` change

### Description
Compilation error on latest `dev`, https://github.com/librespot-org/librespot/commit/eb7c65e77b60da6544e2f4c1bbf3ba06705ef1b7. This is broken since https://github.com/librespot-org/librespot/commit/6f6cd04874b09736bd17879363d1c0a387bc9440 (aka. https://github.com/librespot-org/librespot/pull/1543), the commit prior (https://github.com/librespot-org/librespot/commit/df5f957bdd9a29801dd16f08cdc9abc580912da7) to this change is fine.

### Version
https://github.com/librespot-org/librespot/commit/eb7c65e77b60da6544e2f4c1bbf3ba06705ef1b7

### How to reproduce

Run:

```bash
cargo build  --no-default-features --features "rodio-backend with-avahi native-tls" --release
```

Fails with:

```
   Compiling librespot-discovery v0.7.1 (/home/meril/librespot/discovery)
error[E0433]: failed to resolve: could not find `poll` in `futures_util`
   --> discovery/src/lib.rs:201:60
    |
201 |             while let Poll::Ready(Some(_)) = futures_util::poll!(stream.next()) {
    |                                                            ^^^^ could not find `poll` in `futures_util`

For more information about this error, try `rustc --explain E0433`.
error: could not compile `librespot-discovery` (lib) due to 1 previous error
```

### Host:
- OS: Arch Linux
- Platform: x64
- `rustup show`

```
installed toolchains
--------------------
stable-x86_64-unknown-linux-gnu (active, default)
```

- `rustc --version`

```
rustc 1.90.0 (1159e78c4 2025-09-14)
```

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -89,6 +89,13 @@ jobs:
           --include-features rustls-tls-native-roots
           --exclude-features native-tls,rustls-tls-webpki-roots
 
+      - name: Check discovery features (Linux only)
+        if: runner.os == 'Linux'
+        run: >
+          cargo hack check -p librespot-discovery --each-feature --exclude-all-features
+          --include-features native-tls
+          --exclude-features rustls-tls-native-roots,rustls-tls-webpki-roots
+
       - name: Build binary with default features
         run: cargo build --frozen
 

--- a/discovery/Cargo.toml
+++ b/discovery/Cargo.toml
@@ -13,7 +13,7 @@ edition.workspace = true
 default = ["with-libmdns", "native-tls"]
 
 # Discovery backends
-with-avahi = ["dep:serde", "dep:zbus"]
+with-avahi = ["dep:serde", "dep:zbus", "futures-util/async-await-macro"]
 with-dns-sd = ["dep:dns-sd"]
 with-libmdns = ["dep:libmdns"]
 

